lte/gateway/c/oai/test/pipelined_client/utility_protobuf.h

```diff
@@ -0,0 +1,237 @@
+    }
+
+    return 0;
+  }
+
+  bool validate_req_msg(UESessionSet request) {
+    char req_enb_ip_str[INET_ADDRSTRLEN];
+    char req_ue_ip_str[INET_ADDRSTRLEN];
+
+    inet_ntop(
+        AF_INET, request.enb_ip_address().address().c_str(), req_enb_ip_str,
+        INET_ADDRSTRLEN);
+
+    if (std::strcmp(req_enb_ip_str, enb_v4.c_str())) {
+      return false;
+    }
+
+    inet_ntop(
+        AF_INET, (request.ue_ipv4_address().address().c_str()), req_ue_ip_str,
+        INET_ADDRSTRLEN);
+
+    if (std::strcmp(req_ue_ip_str, ue_v4.c_str())) {
+      return false;
+    }
+
+    if (in_teid != request.in_teid()) {
+      return false;
+    }
+
+    if (out_teid != request.out_teid()) {
+      return false;
+    }
+
+    if (ue_state != get_config_ue_state(request.ue_session_state())) {
+      return false;
+    }
+
+    if (vlan != request.vlan()) {
+      return false;
+    }
+
+    return true;
+  }
+};
+
+class FlowDLOps {
+ public:
+  struct ip_flow_dl flow_dl;
+
+  // Constructor definition
+  FlowDLOps() {
+    set_params   = (DST_IPV4 | SRC_IPV4);
+    tcp_dst_port = 5002;
+    tcp_src_port = 60;
+    udp_dst_port = 0;
+    udp_src_port = 0;
+    ip_proto     = 6;
+    dst_v4       = "192.168.60.141";
+    src_v4       = "192.168.128.11";
+  }
+
+  struct ip_flow_dl get_flow_dl() {
+    return flow_dl;
+  }
+
+  void set_flow_dl() {
+    flow_dl.set_params   = set_params;
+    flow_dl.tcp_dst_port = tcp_dst_port;
+    flow_dl.tcp_src_port = tcp_src_port;
+    flow_dl.udp_dst_port = udp_dst_port;
+    flow_dl.udp_src_port = udp_src_port;
+    flow_dl.ip_proto     = ip_proto;
+    inet_pton(AF_INET, dst_v4.c_str(), &(flow_dl.dst_ip));
+    inet_pton(AF_INET, src_v4.c_str(), &(flow_dl.src_ip));
+  }
+
+  bool validate_flow_dl(IPFlowDL req_flow_dl) {
+    IPAddress* dst_ip_addr;
+    char flow_dl_dst_addr[INET_ADDRSTRLEN];
+    IPAddress* src_ip_addr;
+    char flow_dl_src_addr[INET_ADDRSTRLEN];
+
+    dst_ip_addr = req_flow_dl.mutable_dest_ip();
+    inet_ntop(
+        AF_INET, (dst_ip_addr->address().c_str()), flow_dl_dst_addr,
+        INET_ADDRSTRLEN);
+
+    if (std::strcmp(flow_dl_dst_addr, dst_v4.c_str())) {
+      return false;
+    }
+
+    src_ip_addr = req_flow_dl.mutable_src_ip();
+    inet_ntop(
+        AF_INET, (src_ip_addr->address().c_str()), flow_dl_src_addr,
+        INET_ADDRSTRLEN);
+
+    if (std::strcmp(flow_dl_src_addr, src_v4.c_str())) {
+      return false;
+    }
+
+    if (req_flow_dl.set_params() != set_params) {
+      return false;
+    }
+
+    if (req_flow_dl.tcp_dst_port() != tcp_dst_port) {
+      return false;
+    }
+
+    if (req_flow_dl.tcp_src_port() != tcp_src_port) {
+      return false;
+    }
+
+    if (req_flow_dl.udp_dst_port() != udp_dst_port) {
+      return false;
+    }
+
+    if (req_flow_dl.udp_src_port() != udp_src_port) {
+      return false;
+    }
+
+    if (req_flow_dl.ip_proto() != ip_proto) {
+      return false;
+    }
+
+    return true;
+  }
+
+ private:
+  uint32_t set_params;
```

Comment:
all class private members should have trailing underscore (`set_params_`)

Reply:
Comments addressed.